packages/compiler/core/checker.ts

```diff
@@ -2136,12 +2136,13 @@ export function createChecker(program: Program): Checker {
       type.namespace?.models.set(type.name, type);
     }
 
-    const inheritedPropNames = new Set(
-      Array.from(walkPropertiesInherited(type)).map((v) => v.name)
-    );
+    const inheritedProperties = new Map<string, ModelProperty>();
+    Array.from(walkPropertiesInherited(type)).forEach((item) => {
```

Comment:
Does this work right when the base model shadows a property from superclass? I suppose due to the transitive property of the subtype relationship it should, but it seems like properties higher up the base class list will "win" in this list which feels risky.

Reply:
Could you provide an example? Are you talking about something like A extends B and C extends B?

Reply:
Yeah exactly, something like:

```
model A {
  prop: Type
}

model B extends A {
  prop: Subtype
}

model C extends B {
  prop: SubSubtype
}
```

I think this algorithm will compare against `Type` not `Subtype` when asking if C's prop is a valid subtype.

Reply:
I created this relation:
```
model A {
  prop: int64;
}

model B extends A {
  prop: int32;
}

model C extends B {
  prop: int16;
}
```

This works, as I would expect, but if I change B.prop to say `int8`, the compiler is still happy, even though I would expect it to fail (since int16 won't fit in int8). However, it's clear that the compiler is comparing C.prop (int16) to A.prop (int64). Is this the scenario you are concerned with?

Reply:
> Yeah that's it in general (though I'm unsure if we have any subtype relationships between the sized integer types).

I think we must because if I try to make A.prop int8 it does fail.